src/platform/platform_linux.cpp

```diff
@@ -181,6 +193,57 @@ bool mp::platform::Platform::link(const char* target, const char* link)
     return ::link(target, link) == 0;
 }
 
+QDir mp::platform::Platform::get_alias_scripts_folder()
+{
+    QDir aliases_folder;
+
+    if (mu::in_multipass_snap())
+    {
+        aliases_folder = QDir(QString(mu::snap_user_common_dir()) + "/bin");
+    }
+    else
+    {
+        QString location = MP_STDPATHS.writableLocation(mp::StandardPaths::AppLocalDataLocation) + "/bin";
+        aliases_folder = QDir{location};
+    }
+
+    return aliases_folder;
+}
+
+void mp::platform::Platform::create_alias_script(const std::string& alias, const mp::AliasDefinition& def)
+{
+    std::string file_path = get_alias_script_path(alias);
+
+    std::string multipass_exec = mu::in_multipass_snap() ? "exec /usr/bin/snap run multipass"
+                                                         : QCoreApplication::applicationFilePath().toStdString();
+
+    try
+    {
+        std::string script = "#!/bin/sh\n\n" + multipass_exec + " " + alias + "\n";
+
+        MP_UTILS.make_file_with_content(file_path, script);
+    }
+    catch (const std::runtime_error&)
```

Comment:
Again, shouldn't we be more verbose on what the actual error is?

Reply:
Done. Not catching here lets the exception from `create_alias_script` propagate.